src/api_keys/api.rs

```diff
@@ -78,16 +79,27 @@ pub async fn make_request(
     let url = user_make_request_info
         .api_url
         .replace("xxxREPLACE_MExxx", &api_key_info);
+
+    let mut headers = HeaderMap::new();
+    for (key, value) in &user_make_request_info.http_headers {
+        let first = key.replace("xxxREPLACE_MExxx", &api_key_info);
+        let second = value.replace("xxxREPLACE_MExxx", &api_key_info);
+
+        let header_name = HeaderName::from_bytes(first.as_bytes()).unwrap();
+        let header_value = HeaderValue::from_str(&second).unwrap();
```

Comment:
This one we should handle as some special characters are not allowed in header values which would cause this to panic